src/internal/observable/combineLatest.ts

```diff
@@ -47,6 +48,12 @@ export function combineLatest<A extends readonly unknown[]>(
 
 // combineLatest({a, b, c})
 export function combineLatest(sourcesObject: { [K in any]: never }): Observable<never>;
+/**
+ * You have passed `any` here, we can't figure out if it is
+ * an array or an object, so you're getting `unknown`. Use better types.
+ * @param arg Something typed as `any`
+ */
+export function combineLatest<T extends AnyCatcher>(arg: T): Observable<unknown>;
```

Comment:
suggestion: I would make this the first signature. IDK why `any` skips all of the signatures above this one and I would not like to rely upon that behaviour never changing. There should be no reason this signature cannot be the first.

Reply:
That's a really good idea!